src/Components/Endpoints/src/Builder/RootComponentMetadata.cs

```diff
@@ -0,0 +1,24 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Components.Endpoints.Builder;
+
+/// <summary>
+/// Metadata that represents the root component associated with an endpoint.
+/// </summary>
+public class RootComponentMetadata
```

Comment:
What do we think about the `Microsoft.AspNetCore.Components.Endpoints.Builder` namespace? Do we expect much user code to reference it? If not, it's probably better than `Microsoft.AspNetCore.Builder` namespace `ComponentTypeMetadata` is currently in. But I don't think anything else exists in this namespace yet. What about [Microsoft.AspNetCore.Components.Endpoints](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.endpoints?view=aspnetcore-8.0)? I don't think this deserves its own namespace.

Stuff like the namespace can easily get missed if we try to cram everything in one big API review. Hopefully reviewing this will be fairly quick.